examples/client.ts

```diff
@@ -0,0 +1,74 @@
+import * as Z from "../mod.ts";
+import * as U from "../util/mod.ts";
+
+class ClientConnectError extends Error {
+  override readonly name = "ClientConnectError";
+}
+class ClientDisconnectError extends Error {
+  override readonly name = "ClientDisconnectError";
+}
+
+/** Fake client (to mock interaction with interfaces such as `WebSockets`) */
+class InnerClient {
+  constructor(readonly discoveryValue: string) {}
+
+  close = () => {
+    return new Promise<void>((resolve) => {
+      setTimeout(resolve, 1000);
```

Comment:
Added 5000ms timeout to check the message order and with the following patch

```
diff --git a/examples/client.ts b/examples/client.ts
index 34fb9b8..be0d7b7 100644
--- a/examples/client.ts
+++ b/examples/client.ts
@@ -14,7 +14,7 @@ class InnerClient {
 
   close = () => {
     return new Promise<void>((resolve) => {
-      setTimeout(resolve, 1000);
+      setTimeout(resolve, 5000);
     });
   };
 
@@ -44,7 +44,7 @@ function client<Url extends Z.$<string>>(url: Url) {
       if (false as boolean) {
         return new ClientDisconnectError();
       }
-      return client.close();
+      return client.close().then(() => console.log("DISCONNECTED"));
     },
   );
 }

```

The logs output is 
```
❯ deno task run examples/client.ts
Warning deno task is unstable and may drastically change in the future
Task run deno run -A --no-check=remote "examples/client.ts"
Check file:///Users/parity/ghq/github.com/paritytech/zones/examples/client.ts
CONNECT
DISCONNECT
DISCONNECTED
CALL
{ methodName: "someMethod", args: [ 1, 2, 3 ] }
```

I believe that the expected output should be
```
CONNECT
CALL
{ methodName: "someMethod", args: [ 1, 2, 3 ] }
DISCONNECT
DISCONNECTED
```

Reply:
I've just saw the PR description about exits triggered too soon, so this might be related to that

Reply:
It is